### ISSUE TITLE
Fine Tuning

### ISSUE THREAD

**Issue by NuraddinQJ** [jthickstun/anticipation#7]
Thank you very much for haring this work!!

Is there a way to fine tune autoregressive transformer models? I looked it up but couldn't find a method > my goal is to generate in a specific style on your trained model.

**Comment by jthickstun** [comment]
Hi, sorry for a belated reply; I keep meaning to provide detailed instructions on how to fine-tune a model but it keeps slipping from my schedule.

The current answer is that you can follow the instructions in the train directory for training a model but, instead of training from scratch, hack Levanter to load a pretrained model. You'd also want to change the number of training steps (e.g., 1k-10k steps) in the config, and use a much smaller learning rate (e.g., 1/10 the current max learning rate).

I have been doing some finetuning of these models myself, and will try to provide an example & documentation of how to do this over the next couple months.

**Comment by jthickstun** [comment]
Levanter has changed things around a bit since I posted the previous comment. The vocabulary check hack mentioned in my previous post is no longer needed, but a new small hack required to get things working with the current Levanter HEAD. You need to skip [this check](https://github.com/stanford-crfm/levanter/blob/465591c45a819ca91a381801eb9ff3357a418f35/src/levanter/data/text.py#L353) and instead set `should_append_eos = False` and `should_append_bos = False`.

I'm working with the Levanter team so that things will work out-of-the-box for training music models in the near future.

**Comment by MikeMpapa** [comment]
@jthickstun I also want to thank you for sharing the models and all this documentation with the community. It's a great work and deserves great appreciation !
